functional/util/util.go

```diff
@@ -136,3 +137,20 @@ func TempUnit(contents string) (string, error) {
 
 	return svc, nil
 }
+
+func WaitForState(stateCheckFunc func() bool) (time.Duration, error) {
+	timeout := 15 * time.Second
+	alarm := time.After(timeout)
+	ticker := time.Tick(250 * time.Millisecond)
+	for {
+		select {
+		case <-alarm:
+			// Generic message. Callers can build more specific ones using the supplied timeout value.
```

Comment:
What I meant is that the value is supplied to the *caller* by this function, in the form of a return value... I see though how this wording is confusing. For want of a better idea, I guess I'll just s/supplied/returned/